oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -417,8 +456,28 @@ public boolean equals(Object obj) {
         && Objects.equals(this.scopes, other.scopes);
   }
 
-  private void readObject(ObjectInputStream input) throws IOException, ClassNotFoundException {
-    input.defaultReadObject();
-    transportFactory = newInstance(transportFactoryClassName);
+  protected String createAssertion(JsonFactory jsonFactory, long currentTime) throws IOException {
```

Comment:
`protected` is appropriate for when you want to allow a subclass to override the method, but you want to prevent code outside the subclass & outside the package from calling it; I don't think that's what you're intending though - it looks like you want to make it visible to the test. If that's the case, use package-private.